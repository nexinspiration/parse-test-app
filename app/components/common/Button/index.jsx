import React from 'react';

import styles from '../../../styles/app.css';

export default function Button(props) {

  const handleKeyPress = (event) => {
    // On Enter Press (13), consider it as Button Click
    if (event.which === 13) {
      props.onClick();
    }
  }

  return (
    <button
      className={styles.btn}
      onClick={() => {props.onClick()}}
      onKeyPress={(event) => {handleKeyPress(event)}}>
      {props.text || 'Execute'}
    </button>
  );
}