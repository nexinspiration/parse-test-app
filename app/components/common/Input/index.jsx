import React from 'react';

const className = {
  width: '100%',
  padding: '5px 12px',
  height: '21px',
  border: '1px solid #ccc',
  borderRadius: '2px',
  fontSize: '15px'
};

export default function Input(props) { 
  let inputStyle = Object.assign(className, props.className || {});

  return (
    <input
     style={inputStyle}
     value={props.text}
     onChange={(event) => {props.onChange(event.target.value)}}
    />
  );
}