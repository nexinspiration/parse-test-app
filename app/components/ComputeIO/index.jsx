import React from 'react';

const { Component } = React;

import Input from '../common/Input';

import Button from '../common/Button';

import { isClosed } from '../../utils/common.js'

import styles from '../../styles/app.css';

export class Output extends Component {
  render() {
    return (
      <div className={styles.outputArea}>
        {this.props.text}
      </div>
    );
  }
}

export class ComputeIO extends Component {
  constructor() {
    super();
    this.state = {
      input: '',
      output: ''
    };
  }

  computeOutput() {
    this.setState({
      output: `${isClosed(this.state.input)}`
    });
  }

  render() {
    return (
      <section style={{textAlign: 'left', fontSize: '16px'}}>
        <div style={{marginBottom: '10px'}}>Input:</div>
        <Input text={this.state.input} onChange={(input) => {this.setState({input})}}/>
        <Button onClick={() => this.computeOutput()}/>
        <div style={{marginBottom: '10px'}}>Output:</div>
        <Output text={this.state.output}/>
      </section>
    );
  }
}
