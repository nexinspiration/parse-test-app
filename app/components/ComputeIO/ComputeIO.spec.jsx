import React from 'react';
import { shallow, mount } from 'enzyme';
import { ComputeIO, Output } from '../ComputeIO';
import Input from '../common/Input';
import Button from '../common/Button';
import sinon from 'sinon';
import renderer from 'react-test-renderer';

describe('ComputeIO', () => {
  it('should return an element.', () => {
    const wrapper = shallow(<ComputeIO />);
    expect(wrapper.exists()).toBe(true);
  });

  it('should contain Input, Output & Button elements', () => {
    const wrapper = shallow(<ComputeIO />);
    expect(wrapper.find(Input)).toHaveLength(1);
    expect(wrapper.find(Button)).toHaveLength(1);
    expect(wrapper.find(Output)).toHaveLength(1);
  });

  it('Output computation should trigger on button click', () => {
    sinon.spy(ComputeIO.prototype, 'computeOutput');
    const wrapper = mount(
      <ComputeIO />
    );
    wrapper.find('button').simulate('click');
    expect(ComputeIO.prototype.computeOutput.calledOnce).toEqual(true);
  });

  // Snapshot tests
  it('Renders Correctly', () => {
    const tree = renderer.create(
      <ComputeIO />
    );
    expect(tree).toMatchSnapshot();
  });
});

describe('Output', () => {
  it('should return an element.', () => {
    const wrapper = shallow(<Output />);
    expect(wrapper.exists()).toBe(true);
  });
});
