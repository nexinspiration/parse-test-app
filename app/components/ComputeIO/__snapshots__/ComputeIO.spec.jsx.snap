// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComputeIO Renders Correctly 1`] = `
<section
  style={
    Object {
      "fontSize": "16px",
      "textAlign": "left",
    }
  }
>
  <div
    style={
      Object {
        "marginBottom": "10px",
      }
    }
  >
    Input:
  </div>
  <input
    onChange={[Function]}
    style={
      Object {
        "border": "1px solid #ccc",
        "borderRadius": "2px",
        "fontSize": "15px",
        "height": "21px",
        "padding": "5px 12px",
        "width": "100%",
      }
    }
    value=""
  />
  <button
    className={undefined}
    onClick={[Function]}
    onKeyPress={[Function]}
  >
    Execute
  </button>
  <div
    style={
      Object {
        "marginBottom": "10px",
      }
    }
  >
    Output:
  </div>
  <div
    className={undefined}
  >
    
  </div>
</section>
`;
